test/common/test/api_utils_test.py

```diff
@@ -0,0 +1,249 @@
+                         len(id_list),
+                         msg='Not all nodes have OBM settings {}:{}'.format(node_count, len(id_list)))
+
+    def test_node_obm_validation(self):
+        """
+        Test of the api_node_select() function
+        This validates that if a node is found to have obm setting using nodes_get_all() it
+        will also have obm setting using api_node_select().
+
+        This tests the validate_obm=True and node_id= parameter combination.
+        """
+        Api().nodes_get_all()
+        node_list = self.__get_data()
+
+        for node in node_list:
+            if node.get('type') == 'compute':
+                if 'obms' in node and len(node['obms']) > 0:
+                    id_list = api_node_select(self.__client, node_id=node['id'], validate_obm=True)
+                    self.assertIn(node['id'], id_list, msg='Node {} OBM settings not detected'.format(node['id']))
+
+    def test_obmMac(self):
+        """
+        Test of the api_node_select() function
+        Validate the a obm mac address will return the correct node id.
+
+        This tests the obm_mac= parameter.
+        """
+        obm_count = 0
+        Api().nodes_get_all()
+        node_list = self.__get_data()
+
+        for node in node_list:
+            if 'sku' in node:
+                Api().nodes_get_obms_by_node_id(identifier=node.get('id'))
+                obm_list = self.__get_data()
+                for obm_entry in obm_list:
+                    if obm_entry['config']:
+                        obm_mac = obm_entry['config'].get('host')
+                        id_list = api_node_select(self.__client, obm_mac=obm_mac)
+                        self.assertEqual(1,
+                                         len(id_list),
+                                         msg='Expected a list size of 1 got {} - node id {}'.format(len(id_list),
+                                                                                                    node['id']))
+                        self.assertIn(node['id'],
+                                      id_list,
+                                      msg='Node id {} with obm mac ({}) not found in Id list'.format(node.get('id'),
+                                                                                                     obm_mac))
+                        obm_count += 1
+        self.assertGreater(obm_count, 0, msg='No nodes with OBM found')
+        
+    def test_all_list_arg_node_type(self):
+        """
+        Test of the api_node_select_from_config() function
+        Validate the number of each node type found using nodes_get_all() is the
+        same as the number found be api_node_select_from_config().
+
+        This tests the node_type= parameter.
+        """
+        type_list = ['compute', 'switch', 'enclosure', 'pdu', 'foo']
+
+        Api().nodes_get_all()
+        node_list = self.__get_data()
+        type_count = 0
+
+        print '\n'
```

Comment:
I'm sorry, I missed this last round. We should be moving to infra-logging for new things.

This would be a straight 'logs.debug('Testing node type %s', type_entry)' I believe.